src/main/scala/mesosphere/mesos/TaskBuilder.scala

```diff
@@ -203,6 +202,10 @@ class TaskBuilder(app: AppDefinition,
 
 object TaskBuilder {
 
+  sealed trait BuildResult
```

Comment:
How about adding `isSuccess` and `isFailure` methods to this trait?  It could clean up some of the use sites where you only care about which projection holds for your value.